test/go-tests/sequencetrigger_test.go

```diff
@@ -0,0 +1,289 @@
+		}
+		if !IsEqual(t, "state.State", "triggered", state.State) {
+			return false
+		}
+
+		if !IsEqual(t, "len(state.Stages)", 1, len(state.Stages)) {
+			return false
+		}
+
+		stage := state.Stages[0]
+
+		if !IsEqual(t, "stage.Name", stageName, stage.Name) {
+			return false
+		}
+
+		if !IsEqual(t, "stage.LatestEvent.Type", keptnv2.GetTriggeredEventType("mytask"), stage.LatestEvent.Type) {
+			return false
+		}
+
+		return true
+	}, 10*time.Second, 2*time.Second)
+
+	// get mytask.triggered event of second iteration
+	myTaskTriggeredEvent, err = GetLatestEventOfType(keptnContextID, projectName, stageName, keptnv2.GetTriggeredEventType("mytask"))
+	require.Nil(t, err)
+	require.NotNil(t, myTaskTriggeredEvent)
+
+	cloudEvent = keptnv2.ToCloudEvent(*myTaskTriggeredEvent)
+
+	keptn, err = keptnv2.NewKeptn(&cloudEvent, keptncommon.KeptnOpts{EventSender: &APIEventSender{}})
+	require.Nil(t, err)
+	require.NotNil(t, keptn)
+
+	// send .started event
+	_, err = keptn.SendTaskStartedEvent(nil, source)
+	require.Nil(t, err)
+
+	// send .finished event with result = pass
+	_, err = keptn.SendTaskFinishedEvent(&keptnv2.EventData{
+		Result: keptnv2.ResultPass,
+	}, source)
+	require.Nil(t, err)
+
+	// verify state -> now the next task should have been triggered again
+	require.Eventually(t, func() bool {
+		states, _, err := getState(projectName)
+		if err != nil {
+			return false
+		}
+		stage := states.States[0].Stages[0]
+
+		if !IsEqual(t, "stage.Name", stageName, stage.Name) {
+			return false
+		}
+
+		if !IsEqual(t, "stage.LatestEvent.Type", keptnv2.GetTriggeredEventType("othertask"), stage.LatestEvent.Type) {
+			return false
+		}
+
+		return true
+	}, 10*time.Second, 2*time.Second)
+
+	// get othertask.triggered event of second iteration
+	otherTaskTriggeredEvent, err := GetLatestEventOfType(keptnContextID, projectName, stageName, keptnv2.GetTriggeredEventType("othertask"))
+	require.Nil(t, err)
+	require.NotNil(t, otherTaskTriggeredEvent)
+
+	cloudEvent = keptnv2.ToCloudEvent(*otherTaskTriggeredEvent)
+
+	keptn, err = keptnv2.NewKeptn(&cloudEvent, keptncommon.KeptnOpts{EventSender: &APIEventSender{}})
+	require.Nil(t, err)
+	require.NotNil(t, keptn)
+
+	// send .started event
+	_, err = keptn.SendTaskStartedEvent(nil, source)
+	require.Nil(t, err)
+
+	// send .finished event with result = fail
+	_, err = keptn.SendTaskFinishedEvent(&keptnv2.EventData{
+		Result: keptnv2.ResultFailed,
+	}, source)
+	require.Nil(t, err)
+
+	// verify state -> now the sequence should be finished and not re-triggered again
+	require.Eventually(t, func() bool {
+		states, resp, err := getState(projectName)
+		if err != nil {
+			return false
+		}
+		if !IsEqual(t, "resp.Response().StatusCode", http.StatusOK, resp.Response().StatusCode) {
+			return false
+		}
+		state := states.States[0]
+
+		if !IsEqual(t, "state.State", "finished", state.State) {
+			return false
+		}
+
+		if !IsEqual(t, "len(state.Stages)", 1, len(state.Stages)) {
+			return false
+		}
```

Comment:
It's super unintuitive to read this code here. It's in the reverse order of what stretchr/testify assert would do, e.g.
```golang
  // assert equality
  assert.Equal(t, 123, 123, "they should be equal")
```

Why not use `` here? According to the docs it should work:
```golang
		if !assert.Equal(t, "finished", state.State, "state.State") {
			return false
		}
```